sqlair.go

```diff
@@ -87,64 +89,122 @@ func (db *DB) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 	}
 
 	qe, err := s.pe.Query(inputArgs...)
-	return &Query{qs: db.db, qe: qe, err: err, ctx: ctx}
+	return &Query{qs: db.db, qe: qe, ctx: ctx, err: err}
 }
 
-// Run will execute the query.
-// Any rows returned by the query are ignored.
+// Run is an alias for Get that takes no arguments.
 func (q *Query) Run() error {
+	return q.Get()
+}
+
+// Query.Get will run the query.
+// The first row (if there is one) will be scanned into the outputArgs.
+// It will give ErrNoRows if the query contains output expressions and there are no rows.
+// A pointer to an Outcome struct can be passed as the first argument which will be populated with the outcome of the query.
+func (q *Query) Get(outputArgs ...any) error {
 	if q.err != nil {
 		return q.err
 	}
-	_, err := q.qs.ExecContext(q.ctx, q.qe.QuerySQL(), q.qe.QueryArgs()...)
-	if err != nil {
-		return err
+	var outcome *Outcome
+	if len(outputArgs) > 0 {
+		if oc, ok := outputArgs[0].(*Outcome); ok {
+			outcome = oc
+			outputArgs = outputArgs[1:]
+		}
 	}
-	return nil
+
+	iter := q.Iter()
+	defer iter.Close()
+	if outcome != nil {
+		err := iter.Get(outcome)
+		if err != nil {
+			return err
+		}
+	}
+	if !q.qe.HasOutputs() {
+		if len(outputArgs) > 0 {
+			return fmt.Errorf("cannot get results: query does not return any results")
+		}
+		return iter.Close()
+	}
+	if !iter.Next() {
+		if err := iter.Close(); err != nil {
+			return err
+		}
+		return ErrNoRows
+	}
+	return iter.Get(outputArgs...)
```

Comment:
In this version we would return `ErrNoRows` when they run a query with no output expressions, I've edited it a bit to make it more concise though.

The reason it ends up being so complicated is because we're effectively trying to handle two quite different control flows at the same time, we have the case where there are no output arguments and the case where they want one row.